packages/beacon-node/src/chain/chain.ts

```diff
@@ -417,7 +417,7 @@ export class BeaconChain implements IBeaconChain {
           {dontTransferCache: true},
           RegenCaller.restApi
         );
-        return {state, executionOptimistic: isOptimisticBlock(block)};
+        return {state, executionOptimistic: isOptimisticBlock(block), finalized: false};
```

Comment:
if `slot = finalizedBlock.slot` then finalized = true

same to below

Reply:
Ah right but why do we even execute this branch if `slot = finalizedBlock.slot`? Should this be a `>` here
https://github.com/ChainSafe/lodestar/blob/b1f4e7e674a704a7442640d613920d2cce6c2325/packages/beacon-node/src/chain/chain.ts#L408-L409

And then load the finalized state from state archive db?
https://github.com/ChainSafe/lodestar/blob/b1f4e7e674a704a7442640d613920d2cce6c2325/packages/beacon-node/src/chain/chain.ts#L436

Reply:
if `slot === finalizedSlot` we have the finalized state in checkpoint state cache and it's most of the time is not available in db https://github.com/ChainSafe/lodestar/blob/7fadd303fad8679ee2b7462747b9d9fc33dafb23/packages/beacon-node/src/chain/archiver/archiveStates.ts#L48

Reply:
> Then I think the comment request for non-finalized state is misleading. We might wanna add comment for slot === finalizedBlock.slot scenario

I have updated the code now to reflect this which would make the comment a bit redundant